projects/proto/simulation.proto

```diff
@@ -1,149 +1,97 @@
-    required bytes f = 1;
+    pbTiming timing = 8;
 }
 
 // states are yaw, roll, steer, roll rate, steer rate (5)
-message BicycleStateMessage {
+message pbModelState {
     repeated float x = 1;
 }
 
 // auxiliary states are x, y, rear wheel angle, pitch angle (4)
-message BicycleAuxiliaryStateMessage {
+message pbModelAuxiliaryState {
     repeated float x = 1;
 }
 
 // inputs are roll torque, steer torque (2)
-message BicycleInputMessage {
+message pbModelInput {
     repeated float u = 1;
 }
 
 // STM32F4 ADC, DAC are 12-bit
 // maximum rear wheel encoder count is 1152
-message SensorMessage {
-    required uint32 kistler_measured_torque =   1;
-    required uint32 kollmorgen_actual_torque =  2;
-    required uint32 steer_encoder_count =       3;
-    optional uint32 rear_wheel_encoder_count =  4;
+message pbSensor {
+    uint32 kistler_measured_torque = 1;
+    uint32 kollmorgen_measured_torque = 2;
+    uint32 steer_encoder_count = 3;
+    uint32 rear_wheel_encoder_count = 4;
 }
 
-message ActuatorMessage {
-    required uint32 kollmorgen_command_torque = 1;
-}
-
-// symmetric 5x5 matrix
-message SymmetricStateMatrixMessage {
-    repeated float m = 1;
-}
-
-// symmetric 2x2 matrix
-message SymmetricOutputMatrixMessage {
-    repeated float m = 1;
-}
-
-// 5x2 matrix
-message KalmanGainMatrixMessage {
-    repeated float m = 1;
-}
-
-message BicycleKalmanMessage {
-    optional BicycleStateMessage state_estimate =                          1;
-    optional SymmetricStateMatrixMessage error_covariance =                2;
-    optional SymmetricStateMatrixMessage process_noise_covariance =        3;
-    optional SymmetricOutputMatrixMessage measurement_noise_covariance =   4;
-    optional KalmanGainMatrixMessage kalman_gain =                         5;
+message pbActuator {
+    uint32 kollmorgen_command_torque = 1;
 }
 
 // 2x2 matrix
-message SecondOrderMatrixMessage {
+message pbSecondOrderMatrix {
     repeated float m = 1;
 }
 
 // 5x5 matrix
-message StateMatrixMessage {
+message pbStateMatrix {
     repeated float m = 1;
 }
 
 // 5x2 matrix
-message InputMatrixMessage {
+message pbInputMatrix {
     repeated float m = 1;
 }
 
 // 2x5 matrix
-message OutputMatrixMessage {
+message pbOutputMatrix {
     repeated float m = 1;
 }
 
 // 2x2 matrix
-message FeedthroughMatrixMessage {
+message pbFeedthroughMatrix {
     repeated float m = 1;
 }
 
-message BicycleModelMessage {
-    optional float v =                      1;
-    optional float dt =                     2;
-    optional SecondOrderMatrixMessage M =   3;
-    optional SecondOrderMatrixMessage C1 =  4;
-    optional SecondOrderMatrixMessage K0 =  5;
-    optional SecondOrderMatrixMessage K2 =  6;
-    optional StateMatrixMessage A =         7;
-    optional InputMatrixMessage B =         8;
-    optional OutputMatrixMessage C =        9;
-    optional FeedthroughMatrixMessage D =   10;
+message pbModelStateSpace {
+    float v = 1;
+    pbStateMatrix A = 2;
```

Comment:
I understand A, B, C and D are commonly in control theory but since you found such nice names for these object I figured maybe we could use those. Don't care about this one as much because I think both naming schemes are sensible.

Reply:
Need to move v out.